### PR TITLE
Fix compilation errors : merge HDDS-4308 and HDDS-4473 changes into the branch

### PR DESCRIPTION
**Pull request by rakeshadr**
## What changes were proposed in this pull request?

This is to fix compilation errors by merging HDDS-4308 and HDDS-4473 changes to the branch. 
Need to revisit all the newly created "*RequestV1.java" and "*ResponseV1.java" classes and incorporate HDDS-4308 changes into it. Also, KeyManagerImpl class.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-4321

## How was this patch tested?

Existing V1 UT cases.

### CODE REVIEW

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java [hunk 1]
```diff
@@ -2430,20 +2430,19 @@ private void listStatusFindKeyInTableCache(
...
-        if (args.getSortDatanodes()) {
-          sortDatanodeInPipeline(fileStatus.getKeyInfo(), clientAddress);
-        }
+        keyInfoList.add(fileStatus.getKeyInfo());
       }
     }
+    // refreshPipeline flag check has been removed as part of
+    // https://issues.apache.org/jira/browse/HDDS-3658.
+    // Please refer this jira for more details.
+    refreshPipeline(keyInfoList);
```
Comment (mukul1987) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java:2442]: This change seems unrelated to quota changes.

**Review by linyiqun — COMMENTED** [review]
Other than @mukul1987 's comment, volume quota change in OMFileCreateResponseV1 also needed to update,[OMFileCreateResponseV1.java#L82](https://github.com/apache/ozone/blob/HDDS-2939/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/file/OMFileCreateResponseV1.java#L82)

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java (hunk 1 above)
Comment (rakeshadr) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java:2442]: Thanks @mukul1987  for the reviews. Yes, you are right. Actually after rebasing the feature branch, there are compilation errors and this is due to HDDS-4473 changes.

**Comment by rakeshadr** [comment]
> Besides @mukul1987 's comment, volume quota change in OMFileCreateResponseV1 also needed to update,[OMFileCreateResponseV1.java#L82](https://github.com/apache/ozone/blob/HDDS-2939/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/file/OMFileCreateResponseV1.java#L82)

Thanks @linyiqun for the comment. Will update it.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyDeleteRequestV1.java [hunk 2]
```diff
@@ -149,9 +149,7 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
       omBucketInfo = getBucketInfo(omMetadataManager, volumeName, bucketName);
 
       long quotaReleased = sumBlockLengths(omKeyInfo);
-      // update usedBytes atomically.
-      omVolumeArgs.getUsedBytes().add(-quotaReleased);
-      omBucketInfo.getUsedBytes().add(-quotaReleased);
+      omBucketInfo.incrUsedBytes(-quotaReleased);
```
Comment (bharatviswa504) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyDeleteRequestV1.java:152]: https://github.com/apache/ozone/pull/1607#discussion_r529974503
Pls refer this comment

Comment (bharatviswa504) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyDeleteRequestV1.java:152]: SumBlockLengths for OmKeyInfo here when it is directory it will be zero.

Comment (rakeshadr) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyDeleteRequestV1.java:152]: Thanks a lot @bharatviswa504 for reminding this point again. Since this requires more deeper discussion I've created HDDS-4565 jira.  

Like we discussed in offline chat, it helps to unblock other PRs by fixing compilation errors in the branch(it occurred due to the recent rebase). I am changing the jira description as well reflecting the compilation error part.

**Review by bharatviswa504 — COMMENTED** [review]
+1

**Comment by bharatviswa504** [comment]
As this is blocking further PR's and it is only a compilation issue fix, and anyway, quota will be revisited, I am fine with this.

**Comment by rakeshadr** [comment]
Thanks @mukul1987 , @linyiqun , @bharatviswa504 for the reviews. I will merge it.
